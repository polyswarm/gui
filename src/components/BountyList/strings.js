const strings = {
  empty: 'You haven\'t posted any bounties. Click \'+ BOUNTY\' to post one.',
  view: 'View',
  delete: 'Delete',
  resolved: ' (CLOSED)',
  active: ' (ACTIVE)',
  expired: ' (EXPIRED)',
  nct: ' Nectar (NCT)',
  author: "Author: ",
  assertions: "Assertions: ",
  files: "Files: ",
};
export default strings;