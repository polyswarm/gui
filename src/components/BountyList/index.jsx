// Vendor Imports
import React, {Component} from 'react';
import PropTypes from 'prop-types';
// Bounty Management Imports
import Card from '../Card';

// Component Imports
import strings from './strings';

class BountyList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {props: {bounties}} = this;
    return (
      <div className='BountyList'>
        <ul>
          {bounties && bounties.length > 0 && bounties.map((bounty, index) => {
              return (
                <Card
                  key={bounty.guid}
                  onClick={() => this.onBountySelected(index)}>
                  {bounty.guid}
                </Card>
              );
          })}
        </ul>
        {(!bounties || bounties.length === 0) && (
          <div className='BountyList-Placeholder'>
             <h3>
               {strings.empty}
             </h3>
          </div>
        )}
      </div>
    );
  }

  onBountySelected(index) {
    const {props: {onBountySelected}} = this;
    if (onBountySelected) {
      onBountySelected(index);
    }
  }
}
BountyList.proptypes = {
  bounties: PropTypes.array,
  onBountySelected: PropTypes.func,
}
export default BountyList;