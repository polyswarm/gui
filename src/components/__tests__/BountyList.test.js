import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import BountyList from '../BountyList';

it('renders without crashing', () => {
  const wrapper = render(<BountyList />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('shows all bounties passed as cards', () => {
  const bounties = [
    {guid: 'asdf'},
    {guid: 'fdsa'}
  ];
  const wrapper = mount(<BountyList bounties={bounties}/>);

  expect(wrapper.find('Card')).toHaveLength(2);
  expect(wrapper.find('Card').first().text()).toBe('asdf');
  expect(wrapper.find('Card').last().text()).toBe('fdsa');
});

it('selects the bounty when the card is clicked', () => {
  const onBountySelected = jest.spyOn(BountyList.prototype, 'onBountySelected');
  const bounties = [
    {guid: 'asdf'},
    {guid: 'fdsa'}
  ];
  const wrapper = mount(<BountyList bounties={bounties}/>);
  wrapper.find('Card').first().simulate('click');

  expect(onBountySelected).toHaveBeenCalledWith(0);
});

it('calls prop onSelectBounty when card is clicked', () => {
  const onBountySelected = jest.fn();
  const bounties = [
    {guid: 'asdf'},
    {guid: 'fdsa'}
  ];
  const wrapper = mount(<BountyList bounties={bounties} onBountySelected={onBountySelected}/>);
  wrapper.find('Card').first().simulate('click');

  expect(onBountySelected).toHaveBeenCalledWith(0);
});