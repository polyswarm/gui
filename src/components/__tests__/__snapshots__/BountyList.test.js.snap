// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="BountyList"
>
  <header
    class="Header"
  >
    <div
      class="Header-Row"
    >
      <div
        class="Header-Title"
      >
        <img
          alt="PolySwarm logo"
          class="Header-Logo"
          src="../public/img/polyswarm-white.svg"
        >
          
        </img>
        <h3>
          Bounties & Offers
        </h3>
        <span
          class="RequestSpinner"
        >
          
        </span>
      </div>
      <div
        class="Header-Actions"
      >
        <button
          class="Button header"
        >
          + Bounty
        </button>
        <button
          class="Button header"
        >
          Relay
        </button>
        <div
          class="Dropdown"
        >
          <img
            alt="menu"
            class="Dropdown-Icon"
            height="24px"
            src="../public/img/menu_light.svg"
            width="7px"
          >
            
          </img>
          <div
            class="Dropdown-Choices"
          >
            <p>
              Change Wallet
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Header-Address"
    >
      <p
        class="Header-Divided"
        style="font-size:7.0588235294117645vw"
      >
        author
      </p>
      <p
        class="Header-Divided"
        style="font-size:7.0588235294117645vw"
      >
        1 NCT 
      </p>
      <p
        style="font-size:7.0588235294117645vw"
      >
        1 ETH
      </p>
    </div>
  </header>
  <div
    class="BountyList-Content"
  >
    <ul>
      
    </ul>
    <div
      class="BountyList-Placeholder"
    >
      <h3>
        You haven't posted any bounties. Click '+ BOUNTY' to post one.
      </h3>
    </div>
  </div>
</div>
`;
